File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/setshapeheightandwidth.rb
module Asposediagramjava
  module SetShapeHeightAndWidth
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        shapes = diagram.getPages().getPage(0).getShapes()
        
        i = 0
        while i < shapes.getCount()
            shape = shapes.get(i)
            if shape.getNameU() == "Process" && shape.getID() == 1
                shape.setWidth(2 * shape.getXForm().getWidth().getValue())
                shape.setHeight(2 * shape.getXForm().getHeight().getValue())
            end
            i +=1
        end
        
        # Save diagram
        diagram.save(data_dir + "SetShapeHeightAndWidth.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Set height and width of shape."
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/SetShapeHeightAndWidth.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class SetShapeHeightAndWidth{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram=new Diagram($dataDir."Drawing.vsd");

        $shapes=$diagram->getPages()->getPage(0)->getShapes();

        $i=0;
        while ($i <sizeof($shapes->getCount())) {
            $shape = $shapes->get($i);
            if ((string)$shape->getNameU()=="Process" && (int)(string)$shape->getID()==1) {
            $shape->setWidth(2 * (int)(string)$shape->getXForm()->getWidth()->getValue());
            $shape->setHeight(2 * (int)(string)$shape->getXForm()->getHeight()->getValue());
            }
            $i += 1;
        }
        # Save diagram
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."SetShapeHeightAndWidth.vdx",$saveFileFormat->VDX);
        print "Set height and width of shape.".PHP_EOL;
    }
}